libraries/aws/shadow/src/shadow.c

```diff
@@ -0,0 +1,618 @@
+ */
+#define SHADOW_OP_UPDATE_DELTA                  SHADOW_OP_UPDATE SHADOW_SUFFIX_DELTA
+
+/**
+ * @brief The string representing "/shadow/update/document".
+ */
+#define SHADOW_OP_UPDATE_DOCUMENTS              SHADOW_OP_UPDATE SHADOW_SUFFIX_DOCUMENTS
+
+/**
+ * @brief The string representing "/shadow/delete/accepted".
+ */
+#define SHADOW_OP_DELETE_ACCEPTED               SHADOW_OP_DELETE SHADOW_SUFFIX_ACCEPTED
+
+/**
+ * @brief The string representing "/shadow/delete/accepted".
+ */
+#define SHADOW_OP_DELETE_REJECTED               SHADOW_OP_DELETE SHADOW_SUFFIX_REJECTED
+
+/**
+ * @brief The string representing "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_ACCEPTED                  SHADOW_OP_GET SHADOW_SUFFIX_ACCEPTED
+
+/**
+ * @brief The string representing "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_REJECTED                  SHADOW_OP_GET SHADOW_SUFFIX_REJECTED
+
+/**
+ * @brief The length of "/shadow/update/accepted".
+ */
+#define SHADOW_OP_UPDATE_ACCEPTED_LENGTH               ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/rejected".
+ */
+#define SHADOW_OP_UPDATE_REJECTED_LENGTH               ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/document".
+ */
+#define SHADOW_OP_UPDATE_DOCUMENTS_LENGTH              ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_DOCUMENTS_LENGTH )
+
+/**
+ * @brief The length of "/shadow/update/rejected".
+ */
+#define SHADOW_OP_UPDATE_DELTA_LENGTH                  ( SHADOW_OP_UPDATE_LENGTH + SHADOW_SUFFIX_DELTA_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+#define SHADOW_OP_GET_ACCEPTED_LENGTH                  ( SHADOW_OP_GET_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/rejected".
+ */
+#define SHADOW_OP_GET_REJECTED_LENGTH                  ( SHADOW_OP_GET_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/get/accepted".
+ */
+#define SHADOW_OP_DELETE_ACCEPTED_LENGTH               ( SHADOW_OP_DELETE_LENGTH + SHADOW_SUFFIX_ACCEPTED_LENGTH )
+
+/**
+ * @brief The length of "/shadow/delete/rejected".
+ */
+#define SHADOW_OP_DELETE_REJECTED_LENGTH               ( SHADOW_OP_DELETE_LENGTH + SHADOW_SUFFIX_REJECTED_LENGTH )
+
+/**
+ * @brief Determine if the string contains the substring.
+ *
+ * @param[in] pString Pointer to the string.
+ * @param[in] stringLength Length of pString.
+ * @param[in] pSubString Pointer to the substring.
+ * @param[in] subStringLength Length of pSubString.
+ *
+ * @return Return SHADOW_SUCCESS if it contains;
+ *         return SHADOW_FAIL if not.
+ */
+static ShadowStatus_t containsSubString( const char * pString,
+                                         uint16_t stringLength,
+                                         const char * pSubString,
+                                         uint16_t subStringLength );
+/**
+ * @brief Chek if the Thing Name is valid.
```

Comment:
```suggestion
 * @brief Check if the Thing Name is valid.
```

Reply:
fixed in commit a870748a

resolving this comment